package.json

```diff
@@ -76,6 +76,7 @@
     "jest": "^29.3.1",
     "lint-staged": "^15.0.1",
     "prettier": "^3.0.2",
+    "prettier-2": "npm:prettier@^2.8.8",
```

Comment:
question:

1. Why can't we user the current version of prettier?
2. What does the `npm:` prefix do?

Reply:
I have added relevant reference here:

https://github.com/causaly/zod-validation-error/blob/f46e7fa917383e9ccff511f17d281390cd42a781/jest.config.js#L5-L7

I hope this explains that. 
The root cause that the new tests were failing, because I couldn't get inline snapshots to work with the prettier@v3 we had already installed.

Reply:
tl;dr: Jest isn't compatible with prettier@3 for formatting inline snapshots.

Reply:
Yeah, I saw the comment in the code and have read the link. What I didn't understand is why we needed prettier v2 - you explained it ver well below.

> tl;dr: Jest isn't compatible with prettier@3 for formatting inline snapshots.

However, you didn't explain what `npm:` does. Can you elaborate please?

Also, 

> The root cause was that the new tests were failing, because I couldn't get inline snapshots to work with the prettier@v3 we had already installed.

How is it possible that the tests fail because of prettier?

Reply:
> what npm: does

Actually `npm:` is a sugar syntax, pointing `npm` package manager to understand where to retrieve `prettier-2` from. An alternative of that syntax is the `foo: file:../packages/foo`, where in that case this `file:` sugar syntax points `npm` to install the `foo` package from a relative path.

> How is it possible that the tests fail because of prettier?

Notice that the NEW tests were failing, not all tests. And that's because `jest` is using `prettier` in order to format the inline snapshots, and in our case because of the incompatibility of the `prettier` version it couldn't write the NEW snapshots. The old tests were normally passing.

![Screenshot 2023-10-27 at 9 38 35 AM](https://github.com/causaly/zod-validation-error/assets/22836902/b77133d3-0351-4841-ad5e-0f5fa099f094)

Reply:
Re: `npm:`, couldn't this be written as...

```
"prettier-2": "prettier@^2.8.8",
```

I am just wondering whether `npm:` will be introducing a discrepancy with other modules. 

Finally, should we maybe just drop stupid jest integration with prettier?

Reply:
Rolled back to prettier@v2 to make sure that we don't introduce two different versions of this dev dependency just for the `jest` compatibility.